x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportOpenJobAction.java

```diff
@@ -131,11 +132,14 @@ static void validate(String jobId, Job job) {
 
     static PersistentTasksCustomMetaData.Assignment selectLeastLoadedMlNode(String jobId, Job job,
```

Comment:
This whole method will have to be moved into a place where it can be shared with the data frame analytics allocation code, but I thought I'd save that for a future PR.